databuilder/tables/beta/tpp.py

```diff
@@ -58,6 +59,23 @@ class practice_registrations(EventFrame):
         ],
     )
 
+    def active_for_patient_at(self, date):
+        """
+        Return each patient's "active" practice registration as of the supplied date
+        """
+        spanning_registrations = self.take(
+            self.start_date <= date & ((self.end_date > date) | self.end_date.is_null())
```

Comment:
We should use `.drop` (or whatever we end up calling it) here.

Reply:
Yes, you're right, we should. We end up generating some slightly silly SQL which is why I shied away from it. But I've got plans to fix the silly SQL and we shouldn't let that drive the kind of ehrQL we write.